### PR TITLE
US680194: Unique name space is added to code entities

### PR DESCRIPTION
**Pull request by uppoju**
## Description
Unique namespace is added to policy and encass entities
## Links
Link Type   | Link
------      | ------
Rally Issue | 
Func Spec   | 
## Checklist
- [x] Pull Request Created
- [ ] Code Review Completed
- [ ] Veracode scan results addressed
- [ ] SonarQube scan results addressed
- [ ] TPSR has been submitted (if applicable) 
- [ ] Unit tests have been added
- [ ] Integration/Functional test cases have been written and automated
- [ ] Func Spec has been updated as needed
- [ ] Rally Issue(s) are in an engineering completed state

### CODE REVIEW

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java [hunk 1]
```diff
@@ -58,12 +60,33 @@ public void addDependentBundle(DependentBundle dependentBundle){
...
-    }
+    /**
+     * Applies unique name space and version to the given entity name
+     * ::groupName.bundleName::entityName::majorVersion.minorVersion
+     *
+     * @param entityName String
+     * @return String
+     */
+    public String applyUniqueName(final String entityName) {
+        StringBuilder uniqueName = new StringBuilder(uniqueNameSeparator);
```
Comment (graju256) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java:71]: [optional] think of computing the details based on the project-info only once.

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java [hunk 2]
```diff
@@ -58,12 +60,33 @@ public void addDependentBundle(DependentBundle dependentBundle){
...
+        if (StringUtils.isNotBlank(projectInfo.getGroupName())) {
+            uniqueName.append(projectInfo.getGroupName());
+            uniqueName.append(".");
+        }
+        uniqueName.append(getAnnotatedBundleName());
```
Comment (graju256) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java:76]: I assume shared or [reusable] annotation will be taken care of later. The entity that is shared across multiple bundles shouldn't include bundle names in it.

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java [hunk 3]
```diff
@@ -15,7 +14,7 @@
     private AnnotatedEntity<? extends GatewayEntity> annotatedEntity;
     private ProjectInfo projectInfo;
     private List<DependentBundle> dependentBundles = new ArrayList<>();
-    private String uniqueNameSeparator = "#";   // This can be different for environment entities.
+    private String uniqueNameSeparator = "::";   // This can be different for environment entities.
```
Comment (graju256) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java:17]: You may define it as **final** now.

File: config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundleTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    void testApplyUniqueName() {
+        String uniqueSeparator = "::";
+        String policeName = "TestPolicy";
+        Encass encass = new Encass();
+        AnnotatedEntity<Encass> encassAnnotatedEntity = new AnnotatedEntity<>(encass);
+        encassAnnotatedEntity.setEntityName("TestEncass");
+        ProjectInfo projectInfo = new ProjectInfo("TestName", "TestGroup", "unspecified");
+        AnnotatedBundle annotatedBundle = new AnnotatedBundle(null, encassAnnotatedEntity, projectInfo);
+        String uniquePolicyName = annotatedBundle.applyUniqueName(policeName);
+        Assert.assertEquals(uniqueSeparator + projectInfo.getGroupName() + "."
```
Comment (graju256) [inline config-builder/src/test/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundleTest.java:20]: [optional] Unit tests need to be as simple as possible. Especially, one needs to easily understand what is being tested in the sense, what are inputs and outputs.

Here, I prefer Assert.assertEquals("::TestGroup::TestName::TestEncass", ...

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java (hunk 2 above)
Comment (uppoju) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotatedBundle.java:76]: Yes, we will handle shared entities in other story

**Pull request merged**